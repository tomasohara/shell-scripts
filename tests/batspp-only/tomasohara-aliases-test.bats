#!/usr/bin/env bats
#
# This test file was generated using Batspp
# https://github.com/LimaBD/batspp
#

# Constants
VERBOSE_DEBUG=""
TEMP_DIR="/tmp/batspp-106812"

# Setup function
# $1 -> test name
function run_setup () {
	test_folder=$(echo $TEMP_DIR/$1-$$)
	mkdir --parents "$test_folder"
	cd "$test_folder" || echo Warning: Unable to "cd $test_folder"
}

# Teardown function
function run_teardown () {
	: # Nothing here...
}

@test "test of line 1" {
	run_setup "test-of-line-1"

	# Assertion of line 1
	shopt -s expand_aliases
	print_debug "$(ls *.html)" "$(echo -e 'template.html\n')"
	[ "$(ls *.html)" == "$(echo -e 'template.html\n')" ]

	run_teardown
}

@test "test of line 4" {
	run_setup "test-of-line-4"

	# Assertion of line 4
	shopt -s expand_aliases
	print_debug "$(url-path template.html)" "$(echo -e 'file:////home/tomohara/bin/template.html\n')"
	[ "$(url-path template.html)" == "$(echo -e 'file:////home/tomohara/bin/template.html\n')" ]

	run_teardown
}

# This prints debug data when an assertion fail
# $1 -> actual value
# $2 -> expected value
function print_debug() {
	echo "=======  actual  ======="
	bash -c "echo \"$1\" $VERBOSE_DEBUG"
	echo "======= expected ======="
	bash -c "echo \"$2\" $VERBOSE_DEBUG"
	echo "========================"
}
