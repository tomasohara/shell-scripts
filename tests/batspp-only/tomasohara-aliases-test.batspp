$ ls *.html
template.html

$ url-path template.html
file:////home/tomohara/bin/template.html

#===============================================================================
# TODO: find jupyter way to do divider line (or at least bash)
