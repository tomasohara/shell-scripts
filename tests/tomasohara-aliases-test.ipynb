{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD\n",
    "# ls *.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD\n",
    "# url-path template.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#===============================================================================\n",
    "# TODO: find jupyter way to do divider line (or at least bash)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "\n",
    "## linebr Uses simple bash command to create a series of hypens (-)\n",
    "## OLD\n",
    "# alias linebr=\"printf '%*s\\n' \"${COLUMNS:-$(tput cols)}\" '' | tr ' ' -\"\n",
    "\n",
    "alias linebr=\"printf '%132s\\n' ' ' | tr ' ' '-'\"\n",
    "alias testuserpath=\"sed 's|/home/[^/]\\+/|USER_PATH/|'\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "[ \"$(linebr | grep -o \"-\" | wc -l)\" -eq 132 ]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# ls *.rst\n",
    "[[ -n \"$(ls *.rst)\" &&  \"$(ls *.rst)\" =~ docshtest.rst ]]; echo $?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## OLD\n",
    "# # $ url-path docshtest.rst\n",
    "# # | file:////PATH_TO_SHELL_SCRIPTS/shell-scripts/tests/docshtest.rst\n",
    "# ## This trims the all unnecessary paths from reverse and reverses the trim\n",
    "\n",
    "# ## OLD: url-path docshtest.rst | cut -d ' ' -f 2 | rev | cut -c 1-20 | rev | sed \"s|^|PATH_TO_SHELL_SCRIPTS|\"\n",
    "\n",
    "# # Testuserpath from alias-cd-commands\n",
    "# alias testuserpath=\"sed 's|/home/[^/]\\+/|USER_PATH/|'\"\n",
    "# # alias testuserpath=\"sed 's|/home/[^/]\\+/|USER_PATH/[^/]\\+/|'\"\n",
    "\n",
    "# pwd *.rst | testuserpath\n",
    "# ## Sample Output: USER_PATH/tomProject/shell-scripts/tests\n",
    "\n",
    "# [ \"$(pwd *.rst | testuserpath)\" == \"USER_PATH\\/([^\\/]+\\/)*shell-scripts\\/tests\" ]; echo $?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "# Get the absolute paths of all .rst files in the current directory and replace the home directory path\n",
    "# with \"USER_PATH\" using the defined alias\n",
    "output=$(pwd *.rst | testuserpath)\n",
    "\n",
    "# Check if the resulting path matches the expected pattern\n",
    "[[ \"$output\" =~ USER_PATH/([^/]+/)*shell-scripts/tests ]]; echo $?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
