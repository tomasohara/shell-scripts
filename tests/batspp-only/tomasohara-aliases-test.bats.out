1..2
not ok 1 test-1
# (in test file batspp-only/tomasohara-aliases-test.bats, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== ls *.html
#  ==========
# not.a.url.html
# ========= template.html =========
# template.html
# ============================
not ok 2 test-3
# (from function `test-3-actual' in file batspp-only/tomasohara-aliases-test.bats, line 66,
#  in test file batspp-only/tomasohara-aliases-test.bats, line 53)
#   `test-3-actual ' failed with status 127
# ========== url-path template.html
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-only/tomasohara-aliases-test.bats: line 66: url-path: command not found
